runtimes/spiritnet/src/lib.rs

```diff
@@ -553,6 +553,8 @@ impl delegation::Config for Runtime {
 	type Deposit = constants::delegation::DelegationDeposit;
 }
 
+type CTypeBlockNumberSetOrigin = pallet_collective::EnsureProportionAtLeast<AccountId, TechnicalCollective, 3, 5>;
```

Comment:
i would argue that we should do that using a general vote and not the TC.

Reply:
By general vote you are referring to a referendum or the Council collective? I agree we shouldnt give special power to special entities.

Reply:
referendum

Reply:
I don't see why the general public would be concerned about setting the preimage for a CType hash. Yes, it will probably be a one-off operation, but I don't see why we want to wait 2+ weeks to do something that only has technical motivations?

Reply:
Since the block number is set automatically, you would only need to trust the blockchain, that the number is correct. So trust the chain == trust that polkadot checks everything and the KILT governance(all token holders) is honest and fair.
If you allow  the TC to set the block number you would need to trust them, that they set the correct number. It feels weaker.
Also since this should only be necessary once after the upgrade, i would say it's fine to wait 2 weeks.